### ISSUE TITLE
Add verbose flag

### ISSUE THREAD

**Issue by jamesrichford** [alsatian-test/alsatian#173]
This should go through and output debug level logging when Alsatian is running to better understand what's going on and what could be going wrong.

**Comment by jamesrichford** [comment]
Should be available through CLI and Node

**Comment by Jameskmonger** [comment]
What would this flag do? What exactly would be logged?

**Comment by jamesrichford** [comment]
Useful debugging info, just compiling a list now but including what files were loaded / rejected / ignored due to focus just etc. Not a priority obviously at the moment but just doing some investigation into integration tests that flagged up this would be useful

**Comment by winksaville** [comment]
I suggest there is a the notion of at least two levels of verbosity. Debug logging associated with internal Alsatian code and another level associated with the users of Alsatian code, (i.e. issue #196).

For user level code at least if the `--tap` flag is used then any `console.log` statements do end up being seen. But if you don't use `--tap` you will currently see a blank line in the output as can be seen in #196.

**Comment by jamesrichford** [comment]
Yep @winksaville this is a great point and we want to ensure we can see both, we'll block this on #196 for now as the outcome of that will affect what we do here
